src/Exception/ServerException.php

```diff
@@ -1,12 +0,0 @@
-<?php
-
-namespace Http\Client\Exception;
-
-/**
- * Thrown when a server error (5xx) is encountered
- *
- * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
- */
-final class ServerException extends HttpException
```

Comment:
this means that generic libraries can not catch just client or server exceptions specifically. this makes the distinction in different exceptions pointless. i think we should keep them in here. (and keep the factory logic in HttpException)